src/torrent.cpp

```diff
@@ -8288,8 +8260,16 @@ namespace libtorrent
 
 		int ret = 0;
 
-		std::int64_t const fin_time = finished_time();
-		std::int64_t const download_time = int(active_time()) - fin_time;
+		const seconds32 act_time = active_time();
+		static_assert(std::is_same<
+			decltype(seed_rank_priority_duration), decltype(act_time)>::value,
+			"Need to be same ratio type to prevent runtime conversion");
+
+		// use raw duration values because active time and finish time use same
+		// ratio type and don't need to be converted to a equal ratio
+		// and we don't compare them agains seconds we just use fraction value
+		std::int32_t const fin_time = finished_time().count();
+		std::int32_t const download_time = act_time.count() - fin_time;
```

Comment:
currently it does avoid the division because everthing has same duration ratio of type seconds32. And the compile time type check does test it.

This is the critical expression
`fin_time * 100 / download_time < s.get_int(settings_pack::seed_time_ratio_limit)`

for example 20s finish time and 20s download time and seed_time_ratio_limit 100
if we use a duration ratio 5s per count we would have
`int 4 * 100 / int 4 < s.get_int(100)`

if we use duration it would look like
`int(4*5) * 100 / int(4*5) < s.get_int(100)`
with a performance drop because the calclation is done for each element.
Oh pardon, no you are right it would be
`duration(internal4) * 100 / duration(internal4) < s.get_int(100)`
As long as the types are same duration it would be no performance drop, only some internal ratio checks